lib/git.zsh

```diff
@@ -12,10 +12,27 @@ function __git_prompt_git() {
 # Outputs current branch info in prompt format
 function git_prompt_info() {
   local ref
-  if [[ "$(__git_prompt_git config --get oh-my-zsh.hide-status 2>/dev/null)" != "1" ]]; then
-    ref=$(__git_prompt_git symbolic-ref HEAD 2> /dev/null) || \
-    ref=$(__git_prompt_git rev-parse --short HEAD 2> /dev/null) || return 0
-    echo "$ZSH_THEME_GIT_PROMPT_PREFIX${ref#refs/heads/}$(parse_git_dirty)$ZSH_THEME_GIT_PROMPT_SUFFIX"
+  if [[ "$(command git config --get oh-my-zsh.hide-status 2>/dev/null)" != "1" ]]; then
+    branches=$(command git branch --list -a 2> /dev/null)
+    if [[ "$branches" == "" && "$?" == "0" ]]; then
+        # Look for initialized folders
+        info="<initialized>"
+    elif branch=$(command git symbolic-ref HEAD 2> /dev/null); then
+        # Branch name in the form "<upstream branch name>/<brach name>"
+        ref=$(command git for-each-ref --format='%(upstream:short)' ${branch} 2> /dev/null) || return 0
+        info=$(command echo "${ref#refs/heads/}")
+        if [[ "${info}" == "" ]]; then
+            # Handle for no remote branch
+            info=$(command echo "<no-remote>/$(command git rev-parse --abbrev-ref HEAD)" 2> /dev/null) || return 0
+        fi
+    elif [[ "$(command git tag --points-at 2>/dev/null)" != "" ]]; then
+        # Detached at a tag
+        info=$(command git tag --points-at 2>/dev/null)
```

Comment:
This makes two calls unnecessarily. Also, that command may return more than one tag, so we should use `head -1` or use the `git describe --tags --exact-match` command, which always return only one tag.

Reply:
@mcornella  I couldn't reproduce a case where more than one tag is returned. Could you provide an example?